packages/date-time/src/components/DatePicker/DatePicker.base.tsx

```diff
@@ -311,6 +311,15 @@ export class DatePickerBase extends React.Component<IDatePickerProps, IDatePicke
     this.setState(this._getDefaultState());
   }
 
+  public showDatePickerPopup(): void {
```

Comment:
Just a heads up: this prop would need to be added to `IDatePicker` and only accessed using `componentRef`, rather than `ref`.

We should not assume that `ref` should be accessing public class based APIs, so please make sure to add this to the imperative interface. In function components, `ref` should return the root element, so converting this to a function component would likely cause some churn on customers unless this is only officially accessible through `componentRef` (yes, for now, ref would also return it, but let's call that unsupported until this is functional.)

React in general tends to avoid imperative api in favor of declarative props. Is there a way to achieve this as a prop?

If you absolutely see no other way to do this, then, yes, add to the imperative api surface.

Reply:
I think you are right this needs to be added to IDatePicker